src/energyplus/ForwardTranslator.cpp

```diff
@@ -599,7 +642,11 @@ namespace energyplus {
 
       // get objects by type in sorted order
       std::vector<WorkspaceObject> objects = model.getObjectsByType(iddObjectType);
-      std::sort(objects.begin(), objects.end(), WorkspaceObjectNameLess());
+      if (iddObjectType == IddObjectType::OS_Output_Meter) {
+        std::sort(objects.begin(), objects.end(), OutputMeterSorterPredicate);
+      } else {
+        std::sort(objects.begin(), objects.end(), WorkspaceObjectNameLess());
+      }
```

Comment:
Most meters (except children of Building) are translated here when we loop on iddObjectsToTranslate, so we sort differently then